nuvlaedge/agent/settings.py

```diff
@@ -223,22 +228,33 @@ def _create_client_from_settings(self):
         from nuvlaedge.agent.nuvla.client_wrapper import NuvlaClientWrapper
 
         self._nuvla_client = NuvlaClientWrapper(self.nuvla_endpoint, self.nuvla_endpoint_insecure, self.nuvlaedge_uuid)
+
+        # Handle a special case when the NuvlaEdge credentials are provided as ENV variables. These credentials need
+        # to replace any local session if the configuration match with the stored session and Nuvla after login.
         if self.nuvlaedge_api_key and self.nuvlaedge_api_secret:
             logging.info("Nuvla API keys passed as arguments, these will replace local session")
             self._stored_session.credentials = NuvlaApiKeyTemplate(key=self.nuvlaedge_api_key,
                                                                    secret=self.nuvlaedge_api_secret)
+
             self._nuvla_client.nuvlaedge_credentials = self._stored_session.credentials
 
         if self._stored_session and self._stored_session.credentials:
             self._nuvla_client.nuvlaedge_credentials = self._stored_session.credentials
 
+            _login_success = self._nuvla_client.login_nuvlaedge()
+            _uuids_match = (self._nuvla_client.find_nuvlaedge_id_from_nuvla_session() ==
+                            self._stored_session.nuvlaedge_uuid)
+            if _login_success and _uuids_match:
+                # After logging in if UUID's match, save the session to file
+                self._nuvla_client.save_current_state_to_file()
```

Comment:
Inside `login_nuvlaedge()` it always write to the config file if login succeeded.
So the test for UUID matching will not work as expected.
Is there a need to save to file in `login_nuvlaedge()` ?
If we did login to with the content of this file, I don't think we need to save it.
Also are we ensuring `NUVLA_ENDPOINT` cannot be changed ?

Reply:
You are right. My idea was removing it or making it conditional with an attribute of the login_nuvlaedge function. So I guess that I forgot to take it out after doing some tests.